bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -401,20 +410,25 @@ type LocalEtcd struct {
 	// Environment variables passed using ExtraEnvs will override any existing environment variables, or *_proxy environment variables that kubeadm adds by default.
 	// This option takes effect only on Kubernetes >=1.31.0.
 	// +optional
+	// +kubebuilder:validation:MaxItems=100
 	ExtraEnvs []EnvVar `json:"extraEnvs,omitempty"`
 
 	// serverCertSANs sets extra Subject Alternative Names for the etcd server signing cert.
 	// +optional
+	// +kubebuilder:validation:MaxItems=100
 	ServerCertSANs []string `json:"serverCertSANs,omitempty"`
+
 	// peerCertSANs sets extra Subject Alternative Names for the etcd peer signing cert.
 	// +optional
+	// +kubebuilder:validation:MaxItems=100
 	PeerCertSANs []string `json:"peerCertSANs,omitempty"`
 }
 
 // ExternalEtcd describes an external etcd cluster.
 // Kubeadm has no knowledge of where certificate files live and they must be supplied.
 type ExternalEtcd struct {
 	// endpoints of etcd members. Required for ExternalEtcd.
+	// +kubebuilder:validation:MaxItems=100
```

Comment:
I think we can go down to 50 (even more)

Reply:
Yeah I just went with a high-level because the benefit of restricting this to 10 or something is not that big :D (this field only exists once in the entire CRD)